ZuluContent/Zulu/Spells/Earth/AntidoteSpell.cs

```diff
@@ -1,86 +1,72 @@
     {
-        // Original zuluhotel functionality
-        // cure poisons no matter what
-        // leave target with poison immunity for duration based on skill
-
-        // Current functionality:
-        // Cures poison more or less 100% chance
-        // Planned to add poison resistance when we add a zulu based prots system
-
-
-        public override TimeSpan CastDelayBase
+        public override TimeSpan CastDelayBase => TimeSpan.FromSeconds(0);
+    
+        public override double RequiredSkill => 60.0;
+    
+        public override int RequiredMana => 5;
+        
+        public AntidoteSpell(Mobile caster, Item spellItem) : base(caster, spellItem) { }
+
+        public async Task OnTargetAsync(ITargetResponse<Mobile> response)
         {
-            get { return TimeSpan.FromSeconds(0); }
-        }
+            if (!response.HasValue)
+                return;
 
-        public override double RequiredSkill
-        {
-            get { return 60.0; }
-        }
-
-        public override int RequiredMana
-        {
-            get { return 5; }
-        }
+            var target = response.Target;
+            
+            SpellHelper.Turn(Caster, target);
 
-        public AntidoteSpell(Mobile caster, Item spellItem) : base(caster, spellItem)
-        {
-        }
-
-        public override void OnCast()
-        {
-            Caster.Target = new MobileTarget(this, 10, TargetFlags.Beneficial);
-        }
-
-        public void OnTargetFinished(Mobile from)
-        {
-            FinishSequence();
-        }
-
-        public void OnTarget(Mobile from, Mobile m)
-        {
-            if (!Caster.CanSee(m))
+            if (target.CurePoison(Caster))
             {
-                // Seems like this should be responsibility of the targetting system.  --daleron
-                Caster.SendLocalizedMessage(500237); // Target can not be seen.
-                goto Return;
-            }
-
-            if (!CheckBeneficialSequence(m)) goto Return;
-
-            SpellHelper.Turn(Caster, m);
-
-            var p = m.Poison;
-
-            if (p != null)
-                if (m.CurePoison(Caster))
+                if (Caster != target)
                 {
-                    if (Caster != m)
-                        Caster.SendLocalizedMessage(1010058); // You have cured the target of all poisons!
-
-                    m.SendLocalizedMessage(1010059); // You have been cured of all poisons.
+                    Caster.SendLocalizedMessage(1010058); // You have cured the target of all poisons!
+                    target.SendLocalizedMessage(1010059);  // You have been cured of all poisons.
+                }
+                else
+                {
+                    Caster.SendLocalizedMessage(1010059);  // You have been cured of all poisons.
                 }
 
+                if (target is IEnchanted enchantedTarget && enchantedTarget.Enchantments.GetResist(ElementalType.Poison) == 0)
```

Comment:
@jabinb but if they had poison prot equipped then this check wouldn't pass no? https://github.com/zuluhotelaustralia/zha-2004-2006-side/blob/main/zhc93/pkg/opt/earth/antidote.src#L65

Reply:
I don't think so, unless there's some magic propagating the `PermPoisonImmunity` CProp onto the mobile. In any case for us it's not necessary to check since buffs are separate from the item enchantments and don't overwrite each other.

Reply:
ok done